specification/iotoperations/IoTOperations.Management/models/akri/connectorTemplate.tsp

```diff
@@ -0,0 +1,422 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@azure-tools/typespec-azure-resource-manager";
+import "@azure-tools/typespec-client-generator-core";
+import "./common.tsp";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using Azure.ResourceManager;
+using TypeSpec.Versioning;
+using TypeSpec.OpenAPI;
+using Azure.ClientGenerator.Core;
+
+namespace Microsoft.IoTOperations;
+
+/** AkriConnectorTemplate resource. */
+@parentResource(InstanceResource)
+@added(Versions.`2025-07-01-preview`)
+model AkriConnectorTemplateResource
+  is ProxyResource<AkriConnectorTemplateProperties> {
+  /** Name of AkriConnectorTemplate resource. */
+  @pattern("^[a-z0-9][a-z0-9-]*[a-z0-9]$")
+  @key("akriConnectorTemplateName")
+  @path
+  @minLength(3)
+  @maxLength(63)
+  @segment("akriConnectorTemplates")
+  name: string;
+
+  /** Edge location of the resource. */
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-invalid-envelope-property"
+  @visibility(Lifecycle.Read, Lifecycle.Create)
+  extendedLocation?: ExtendedLocation;
+}
+
+/** AkriConnectorTemplate properties. */
+@added(Versions.`2025-07-01-preview`)
+model AkriConnectorTemplateProperties {
+  /** The status of the last operation. */
+  @visibility(Lifecycle.Read)
+  provisioningState?: ProvisioningState;
+
+  /** Metadata about AIO. */
+  aioMetadata?: AkriConnectorTemplateAioMetadata;
+
+  /** The runtime configuration for the Connector template. */
+  runtimeConfiguration: AkriConnectorTemplateRuntimeConfiguration;
+
+  /** Diagnostics settings for the Connector template. */
+  diagnostics?: AkriConnectorTemplateDiagnostics;
+
+  /** Device inbound endpoint types. */
+  @identifiers(#["endpointType", "version"])
+  @minItems(1)
+  deviceInboundEndpointTypes: AkriConnectorTemplateDeviceInboundEndpointType[];
+
+  /** Mqtt connection configuration settings. */
+  mqttConnectionConfiguration?: AkriConnectorsMqttConnectionConfiguration;
+}
+
+/** AkriConnectorTemplateDeviceInboundEndpointType properties. */
+@added(Versions.`2025-07-01-preview`)
+model AkriConnectorTemplateDeviceInboundEndpointType {
+  /** A description of the device inbound endpoint. */
+  description?: string;
+
+  /** The type of the device inbound endpoint. */
+  endpointType: NonEmptyString;
+
+  /** The version of the device inbound endpoint. */
+  version: NonEmptyString;
```

Comment:
This should be optional. 
@jollyaakash @abhipsaMisra fyi.. pls confirm

Reply:
updated.

Reply:
@abhipsaMisra, can you confirm that making version optional in the RP is the correct change?

Reply:
This is the latest I see from Peri:
> Ok, we're taking the RP change in 2507! This field will be optional

Based on this, it sounds like this is changing in the RP. We will then also need to update the `ConnectorTemplate` CRD so that it is in sync with the RP.
@varunpuranik , @SARSOMA , @jollyaakash FYI

Reply:
yes, that is correct - it should be optional in our CRD as well. 
But we need to coordinate that change.